tests/test_upload_commands.py

```diff
@@ -0,0 +1,200 @@
+"""
+Integration test for upload_commands.
+"""
+
+import dronekit
+
+from modules import upload_commands
+
+
+MAVLINK_FRAME = dronekit.mavutil.mavlink.MAV_FRAME_GLOBAL_RELATIVE_ALT
+MAVLINK_COMMAND = dronekit.mavutil.mavlink.MAV_CMD_NAV_WAYPOINT
+MAVLINK_TAKEOFF = dronekit.mavutil.mavlink.MAV_CMD_NAV_TAKEOFF
+MAVLINK_LANDING = dronekit.mavutil.mavlink.MAV_CMD_NAV_LAND
+DELAY = 3
+CONNECTION_ADDRESS = "tcp:localhost:14550"
+TOLERANCE = 0.00001
+
+
+def test_upload_waypoint_command_list(drone: dronekit.Vehicle,
+                                      commands: "list[dronekit.Command]") -> None:
+    """
+    Test the case of a list of waypoint commands.
+    """
+    upload_commands.upload_commands(drone,commands)
+
+    # Retrieve current drone commands and see if they match with inputs
+    command_sequence = drone.commands
+    command_sequence.download()
+    command_sequence.wait_ready()
+    
+    for i, command in enumerate(command_sequence):
+        lat_input = waypoints_input[i][0]
+        lon_input = waypoints_input[i][1]
+        
+        assert command.frame == MAVLINK_FRAME
+        assert command.command == MAVLINK_COMMAND
+        assert command.param1 == DELAY
+        # Parameters 2,3,4 not being tested since Mission Planner ignores them
+        assert command.param2 == 0 
+        assert command.param3 == 0
+        assert command.param4 == 0
+        assert abs(command.x - lat_input) <= TOLERANCE 
+        assert abs(command.y - lon_input) <= TOLERANCE
+        assert command.z == altitude_input
+
+
+def test_upload_empty_command_list(drone: dronekit.Vehicle,

```

Comment:
To test this, don't pass commands in. Instead:
1. Download the command sequence (do not clear it)
2. Copy the commands into a list
3. Run the upload with an empty list
4. Loop through the command sequence to compare them to the list (you can create a helper function since the other test does the same thing)